fix: Python: Semantic Kernel sample bug fix & updates

moonbox3 opened the pull request:
# Description

The Semantic Kernel Python Agent sample has a bug related to checking for a non-existent `_thread_id`. The underlying SDK was fixed and the sample does not have the change to properly check for the `.id` property on the thread.

Additionally, we fixed the `on_intermediate_message` callback as part of the Semantic Kernel agent framework, which properly emits the `FunctionCallContent` and `FunctionResultContent` during function calling. The sample is now updated to reflect this, along with updated interim messsaging to the host during function call processing.

The PR also includes a few smaller updates in the README to make sure the dev starts the host with the proper port that the SK agent server uses.

Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [X] Follow the [`CONTRIBUTING` Guide](https://github.com/google/A2A/blob/main/CONTRIBUTING.md).
- [X] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
- [X] Ensure the tests and linter pass (Run `nox -s format` from the repository root to format)
- [X] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

moonbox3 commented:
Hi @rajeshvelicheti, @kthota-g thanks in advance for your help getting this bug fix/updates in for the Semantic Kernel agent sample.

kthota-g commented:
@moonbox3 we are updating the repo to the next revision of the protocol. We will uptake this change once the merge is complete.

kthota-g commented:
@moonbox3 `main` has the 0.2.0 changes now and we have updated the example to use the sdk. I assume removing the task_manager from this PR should resolve the conflicts.

moonbox3 commented:
> @moonbox3 `main` has the 0.2.0 changes now and we have updated the example to use the sdk. I assume removing the task_manager from this PR should resolve the conflicts.

Thank you, @kthota-g. I pulled in the latest changes, and removed the `task_manager`. The sample is working well now. Appreciate the help.
